docs/source/02_architecture.md

```diff
@@ -0,0 +1,123 @@
+this allows developers to write at a higher level of abstraction without
+sacrificing performance, since we have the freedom to change the implementation
+of a layer of the stack in order to improve the performance of a given query.
+
+![The ADAM Stack Model](source/img/stack-model.pdf)
+
+This stack model is divided into seven levels. Starting from the bottom,
+these are:
+
+1. The *physical storage* layer is the type of storage media (e.g., hard
+   disk/solid state drives) that are used to store the data.
+2. The *data distribution* layer determines how data is made accessible to all
+   of the nodes in the cluster. Data may be made accessible through a
+   POSIX-compliant shared file system such as NFS [@sandberg85], a non-POSIX
+   file system that implements Hadoop's APIs (e.g., HDFS), or through a
+   block-store, such as Amazon S3.
+3. The *materialized data* layer defines how the logical data in a single
+   record maps to bytes on disk. We advocate for the use of [Apache
+   Parquet](https://parquet.apache.org), a high performance columnar storage
+   format based off of Google's Dremel database [@melnik10]. To support
+   conventional genomics file formats, we exchange the Parquet implementation
+   of this layer with a layer that maps the given schema into a traditional
+   genomics file format (e.g., SAM/BAM/CRAM, BED/GTF/GFF/NarrowPeak, VCF).
+4. The *schema* layer defines the logical representation of a datatype.
+5. The *evidence access* layer is the software layer that implements and
+   executes queries against data stored on disk. While ADAM was originally
+   built around Apache Spark's Resilient Distributed Dataset (RDD)
+   API [@zaharia12], ADAM has recently enabled the use of Spark
+   SQL [@armbrust15] for evidence access and query.
+6. The *presentation* layer provides high level abstractions for interacting
+   with a parallel collection of genomic data. In ADAM, we implement this layer
+   through the [GenomicRDD](#genomic-rdd) classes. This layer presents users
+   with a view of the metadata associated with a collection of genomic data,
+   and APIs for [transforming](#transforming) and [joining](#join) genomic data.
+   Additionally, this is the layer where we provide cross-language support.
+7. The *application* layer is the layer where a user writes their application
+   on top of the provided APIs.
+
+Our stack model derives its inspiration from the layered Open Systems
+Interconnection (OSI) networking stack [@zimmermann80], which eventually served
+as the inspiration for the Internet Protocol stack, and from the concept of data
+independence in database systems. We see this approach as an alternative to the
+"stack smashing" commonly seen in genomics APIs, such as the GATK's "walker"
+interface [@mckenna10]. In these APIs, implementation details about the layout
+of data on disk (is the data sorted?) are propegated up to the application layer
```

Comment:
(is the data sorted?) &rarr; (are the data sorted?)
propegated &rarr; propagated